Neos.Flow/Classes/Security/Authentication/Token/BearerToken.php

```diff
@@ -0,0 +1,63 @@
+<?php
+declare(strict_types=1);
+
+namespace Neos\Flow\Security\Authentication\Token;
+
+/*
+ * This file is part of the Neos.Flow package.
+ *
+ * (c) Contributors of the Neos Project - www.neos.io
+ *
+ * This package is Open Source Software. For the full copyright and license
+ * information, please view the LICENSE file which was distributed with this
+ * source code.
+ */
+
+use Neos\Flow\Mvc\ActionRequest;
+use Neos\Flow\Security\Authentication\TokenInterface;
+use Neos\Flow\Security\Exception\AuthenticationRequiredException;
+use Neos\Flow\Security\Exception\InvalidAuthenticationStatusException;
+
+/**
+ * rfc6750 Bearer token (https://tools.ietf.org/html/rfc6750)
+ *
+ * "A security token with the property that any party in possession of the token (a "bearer")
+ * can use the token in any way that any other party in possession of it can. Using a bearer
+ * token does not require a bearer to prove possession of cryptographic key material (proof-of-possession)."
+ */
+class BearerToken extends AbstractToken implements SessionlessTokenInterface
+{
+    protected $bearer = '';
+
+    /**
+     * @param ActionRequest $actionRequest
+     * @throws AuthenticationRequiredException
+     * @throws InvalidAuthenticationStatusException
+     */
+    public function updateCredentials(ActionRequest $actionRequest)
+    {
+        $this->setAuthenticationStatus(self::AUTHENTICATION_NEEDED);
+        $httpRequest = $actionRequest->getHttpRequest();
+
+        if (!$httpRequest->hasHeader('Authorization')) {
+            return;
+        }
+
+        $authorizationHeader = $httpRequest->getHeader('Authorization')[0];
+
+        if (!str_starts_with($authorizationHeader, 'Bearer ')) {
+            $this->setAuthenticationStatus(TokenInterface::NO_CREDENTIALS_GIVEN);
+            throw new AuthenticationRequiredException('Could not extract access token from Authorization header: "Bearer" keyword is missing', 1616222749);
+        }
+
+        $this->bearer = substr($authorizationHeader, strlen('Bearer '));
```

Comment:
Q: Isn't the expectation of the `AbstractToken` that the `credentials` array property is set? So that `getCredenttials()` will return it later. Otherwise that method needs to be overridden. Also note that the `credentials` property is `@Flow\Transient` - though this token is probably never stored in the database anyway

Reply:
Doh, good point. That would always be empty with this implementation, which is unexpected.

Reply:
Thanks @albe / @kdambekalns. Adjusted the code as suggested.